src/zc/buildout/easy_install.py

```diff
@@ -169,6 +169,60 @@ def _execute_permission():
 
 _easy_install_cmd = 'from setuptools.command.easy_install import main; main()'
 
+def _get_source(filename):
+    if not os.path.isfile(filename):
+        return None
+    with open(filename, 'rb') as f:
+        return f.read()
+
+def get_namespace_package_paths(dist):
+    """
+    Generator of the expected pathname of each __init__.py file of the
+    namespaces of a distribution.
+    """
+    base = [dist.location]
+    init = ['__init__.py']
+    for namespace in dist.get_metadata_lines('namespace_packages.txt'):
+        yield os.path.join(*(base + namespace.split('.') + init))
+
+def get_namespace_package_sources(dist):
+    for path in get_namespace_package_paths(dist):
+        yield _get_source(path)
+
+def dist_needs_pkg_resources(dist):
+    """
+    A distribution needs setuptools/pkg_resources added as requirement if:
+
+        * It has namespace packages declared with:
+        - `pkg_resources.declare_namespace()`
+        * Those namespace packages don't fall back to `pkgutil`
+        * It doesn't have `setuptools/pkg_resources` as requirement already
+    """
+    def requires_pkg_resources_already(dist):
```

Comment:
> Why are this and the following function nested?

Because they're tiny functions that are only used once by another tiny function...?

I like that the final algorithm for deciding if `setuptools` is needed is expressed clearly in the `return` expression, so I inlined those functions that are never going to be used outside of `dist_needs_pkg_resources()` to make that nice looking `return` expression.

As for the rest of the tiny functions, I actually use `get_namespace_package_paths()` in `buildout.wheel` to fix up PEP 420 namespaces for Python 2.

If you'd rather we get rid of the crutch. I can move it over there (and maybe I'll even inline it!).

Otherwise, I can try to inline most of them and keep just this one out.

But in general, IMO Buildout is plagued by very long multi-screen functions and methods, which I think makes them hard to understand. The interaction between `Installer.install()`, `Installer._get_dist()` and `Installer._call_easy_install()` took many passes for me to see that there was a lot of redundant code between them and where I could add the smallest possible wheel hook.

So, between the context switching (which I try to alleviate using expressive function names), and multi-page functions, we need to strike some balance for maximum readability.